Objective-C/Tests/ReplicatorTest.m

```diff
@@ -1855,6 +1855,131 @@ - (void) testPushAndForget {
     [replicator removeChangeListenerWithToken: docReplToken];
 }
 
+
+- (void) testPullRemovedDocWithFilterSingleShot {
+    [self testPullRemovedDocWithFilter: NO];
+}
+
+
+- (void) testPullRemovedDocWithFilterContinuous {
+    [self testPullRemovedDocWithFilter: YES];
+}
+
+
+- (void) testPushRemovedDocWithFilterSingleShot {
+    [self testPushRemovedDocWithFilter: NO];
+}
+
+
+- (void) testPushRemovedDocWithFilterContinuous {
+    [self testPushRemovedDocWithFilter: NO];
+}
+
+
+- (void) testPullRemovedDocWithFilter: (BOOL)isContinuous {
+    NSError* error;
+    CBLMutableDocument* doc1 = [[CBLMutableDocument alloc] initWithID: @"doc1"];
+    [doc1 setString: @"pass" forKey: @"name"];
+    Assert([otherDB saveDocument: doc1 error: &error]);
+    
+    CBLMutableDocument* doc2 = [[CBLMutableDocument alloc] initWithID: @"pass"];
+    [doc2 setString: @"pass" forKey: @"name"];
+    Assert([otherDB saveDocument: doc2 error: &error]);
+    
+    // replicator with pull filter
+    NSMutableSet<NSString*>* docIds = [NSMutableSet set];
+    id target = [[CBLDatabaseEndpoint alloc] initWithDatabase: otherDB];
+    CBLReplicatorConfiguration* config = [self configWithTarget: target
+                                                           type: kCBLReplicatorTypePull
+                                                     continuous: isContinuous];
+    config.pullFilter = ^BOOL(CBLDocument* document, CBLDocumentFlags flags) {
+        AssertNotNil(document.id);
+        [docIds addObject: document.id];
```

Comment:
I think the correct logic should be adding the docIds only when getting the access removed flag?